src/main/java/uk/ac/ebi/ena/readtools/loader/common/eater/AbstractDataConsumer.java

```diff
@@ -18,11 +18,11 @@
 import java.util.Map.Entry;
 
 
-public abstract class 
-AbstractDataEater<T1, T2> implements DataEater<T1, T2>
+public abstract class
+AbstractDataConsumer<T1, T2> implements DataConsumer<T1, T2>

```

Comment:
Remove unused setOk(boolean is_ok). T1 and T2 should be restricted to concrete types (e.g. marker interfaces). We should log rather than write to System.out.

Reply:
This class resides in a common package and is probably meant to be used by anything that considers it useful. A marker interface can be created but I dont think it would be of any help unless it is given a meaningful name and we know what is it useful for exactly. Concrete classes such as IlluminaSingleDataConsumer, IlluminaPairedDataConsumer do define the types when they inherit these abstract classes so i suppose thats good enough.

Reply:
We need some restriction types for the templates otherwise it is impossible for the occasional reader (like me) to figure out the expected relationship of T1 and T2 an the AbstractDataConsumer. AbstractDataConsumer is not a container class that can accept any Object. Because T1 and T2 are restricted we should be explicit about what they are restricted to in the class definion.